src/test/e2e/cypress/e2e/search.cy.js

```diff
@@ -2,97 +2,99 @@ beforeEach(() => {
     cy.visit("/search");
 })
 
-it('should show info message on search page', () => {
-    cy.get('body').then((body) => {
-        expect(body).to.contain("Suche nach einer Prüfung...")
+describe('Verify content of search page', () => {
+    it('should show info message on search page', () => {
+        cy.get('body').then((body) => {
+            expect(body).to.contain("Suche nach einer Prüfung...")
+        })
     })
-})
-
-it('should display header on search page', () => {
-    cy.get('#logo').should("be.visible");
-})
 
-it('should display label for searchbar', () => {
-    cy.get('body').then((body) => {
-       cy.contains("Prüfungsnummer eingeben:")
+    it('should display header image on search page', () => {
+        cy.get('#logo').should("be.visible");
     })
-});
 
-it('should display input for exam number', () => {
-    cy.get('input[type="text"]').should('exist');
-});
-
-it('should display results when searching with valid exam number', () => {
-    cy.get('input[type="text"]').type('11000');
-    cy.get('button').click();
-    cy.contains('Privatrecht');
-});
+    it('should display label for searchbar', () => {
+        cy.get('body').then(() => {
+            cy.contains("Prüfungsnummer eingeben:")
+        })
+    });
 
-it('should display error when no exams were found', () => {
-    cy.get('input[type="text"]').type('99999');
-    cy.get('button').click();
-    cy.contains('Keine Prüfungen für die Prüfungslaufnummer 99999 gefunden');
-});
+    it('should display input for exam number', () => {
+        cy.get('[data-testid="exam-number-input"]').should('exist');
+    });
+})
 
-it('should display error when exam number is too short', () => {
-    cy.get('input[type="text"]').type('123');
-    cy.get('button').click();
-    cy.contains('Prüfungsnummer muss aus genau 5 Ziffern bestehen.');
+describe('Verify searchbar functionality', () => {
+    it('should display results when searching with valid exam number', () => {
+        cy.getByTestId("exam-number-input").type('11000');
+        cy.getByTestId("submit-button").click();
+        cy.contains('Privatrecht');
+    });
+
+    it('should display error when no exams were found', () => {
+        cy.getByTestId("exam-number-input").type('99999');
+        cy.getByTestId("submit-button").click();
+        cy.contains('Keine Prüfungen für die Prüfungslaufnummer 99999 gefunden');
+    });
+
+    it('should display error when exam number is too short', () => {
+        cy.getByTestId("exam-number-input").type('123');
+        cy.getByTestId("submit-button").click();
+        cy.contains('Prüfungsnummer muss aus genau 5 Ziffern bestehen.');
 });
 
 it('should display error when input consists of letters', () => {
     cy.get('input[type="text"]').type('asdf');
     cy.get('[data-testid=\"submit-button\"]').click();
```

Comment:
Is there a reason why ` getByTestId()` is not used here? 
```suggestion
    cy.getByTestId("submit-button").should('exist');
```